source/digits_hits/src/GateRejectedVolume.cc

```diff
@@ -0,0 +1,68 @@
+/*----------------------
+   Copyright (C): OpenGATE Collaboration
+
+This software is distributed under the terms
+of the GNU Lesser General  Public Licence (LGPL)
+See LICENSE.md for further details
+----------------------*/
+
+
+#include "GateRejectedVolume.hh"
+
+#include "G4UnitsTable.hh"
+
+#include "GateRejectedVolumeMessenger.hh"
+#include "GateTools.hh"
+
+GateRejectedVolume::GateRejectedVolume(GatePulseProcessorChain* itsChain,
+			       const G4String& itsName,
+      	      	      	      	 G4String rejectedVolume)
+  : GateVPulseProcessor(itsChain,itsName),
+    m_rejectedVolume(rejectedVolume)
+{
+  m_messenger = new GateRejectedVolumeMessenger(this);
+}
+
+
+
+
+GateRejectedVolume::~GateRejectedVolume()
+{
+  delete m_messenger;
+}
+
+
+
+void GateRejectedVolume::ProcessOnePulse(const GatePulse* inputPulse,GatePulseList& outputPulseList)
+{
+  if (!inputPulse) {
+    if (nVerboseLevel>1)
+      	G4cout << "[GateRejectedVolume::ProcessOnePulse]: input pulse was null -> nothing to do\n\n";
+    return;
+  }
+  if (((inputPulse->GetVolumeID()).GetBottomCreator())->GetObjectName() == "") {
+    if (nVerboseLevel>1)
+      	G4cout << "[GateRejectedVolume::ProcessOneHit]: volume is null for " << inputPulse << " -> pulse ignored\n\n";
+    return;
+  }
+  if ( ((inputPulse->GetVolumeID()).GetBottomCreator())->GetObjectName() != m_rejectedVolume )
+  {
+    GatePulse* outputPulse = new GatePulse(*inputPulse);
+    outputPulseList.push_back(outputPulse);
```

Comment:
I believe you can change this just to ``outputPulseList.push_back(inputPulse);``. If it will fail because ``const`` I think you should instead do const_cast instead.

Reply:
const_cast works for that